### ISSUE TITLE
TSQL Azure Synapse Analytics: Receiving Internal Error

### ISSUE THREAD

**Issue by jpers36** [sqlfluff/sqlfluff#3623]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Running fix on the following code causes the following error:
```
CREATE TABLE #procs
WITH (DISTRIBUTION = HASH([eid])) AS
WITH proc_icd
AS 
( SELECT
    *
  FROM fbp
)
SELECT
   *
FROM
(
   SELECT
	   *
   FROM proc_icd
) sub
;

CREATE TABLE #tem
WITH (DISTRIBUTION = HASH([eid])) AS
SELECT *
FROM
(
	SELECT
		*
	FROM
		a
) b
;
```

```
==== finding fixable violations ====
WARNING    Unable to lint C:\Users\jpersons\Documents\LinterTest.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "C:\users\jpersons\source\repos\sqlfluff\src\sqlfluff\core\linter\runner.py", line 103, in run
    yield partial()
  File "C:\users\jpersons\source\repos\sqlfluff\src\sqlfluff\core\linter\linter.py", line 741, in lint_rendered
    return cls.lint_parsed(
  File "C:\users\jpersons\source\repos\sqlfluff\src\sqlfluff\core\linter\linter.py", line 667, in lint_parsed
    tree, initial_linting_errors, ignore_buff = cls.lint_fix_parsed(
  File "C:\users\jpersons\source\repos\sqlfluff\src\sqlfluff\core\linter\linter.py", line 586, in lint_fix_parsed
    new_tree, _, _ = tree.apply_fixes(
  File "C:\users\jpersons\source\repos\sqlfluff\src\sqlfluff\core\parser\segments\base.py", line 1278, in apply_fixes
    s, before, after = seg.apply_fixes(dialect, rule_code, fixes)
  File "C:\users\jpersons\source\repos\sqlfluff\src\sqlfluff\core\parser\segments\base.py", line 1278, in apply_fixes
    s, before, after = seg.apply_fixes(dialect, rule_code, fixes)
  File "C:\users\jpersons\source\repos\sqlfluff\src\sqlfluff\core\parser\segments\base.py", line 1278, in apply_fixes
    s, before, after = seg.apply_fixes(dialect, rule_code, fixes)
  [Previous line repeated 1 more time]
  File "C:\users\jpersons\source\repos\sqlfluff\src\sqlfluff\core\parser\segments\base.py", line 1329, in apply_fixes
    r = r.__class__(
  File "C:\users\jpersons\source\repos\sqlfluff\src\sqlfluff\core\parser\segments\base.py", line 266, in __init__
    pos_marker = PositionMarker.from_child_markers(
  File "C:\users\jpersons\source\repos\sqlfluff\src\sqlfluff\core\parser\markers.py", line 105, in from_child_markers
    raise ValueError("Attempted to make a parent marker from multiple files.")
ValueError: Attempted to make a parent marker from multiple files.

==== no fixable linting violations found ====
All Finished!
```

### Expected Behaviour

Rule L042 properly lints the file without an internal error

### Observed Behaviour

The fixer threw an internal error.

### How to reproduce

Run sqlfluff fix on the above code

### Dialect

TSQL

### Version

latest main

### Configuration

```
[sqlfluff]
dialect = tsql
large_file_skip_char_limit = 0
rules = L001,L002,L004,L005,L006,L008,L009,L010,L015,L019,L036,L037,L039,L040,L041,L042,L043,L049,L051,L061,L063

[sqlfluff:rules]
max_line_length = 140
comma_style = leading
single_table_references = qualified
unquoted_identifiers_policy = all

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = upper

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = upper

[sqlfluff:rules:L042]
# By default, allow subqueries in from clauses, but not join clauses.
forbid_subquery_in = both

[sqlfluff:rules:L051]
# Which types of JOIN clauses should be fully qualified?
fully_qualify_join_types = both

[sqlfluff:rules:L063]
extended_capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
